### PR TITLE
Dispatch mechanism implementation

### PR DESCRIPTION
**Pull request by mattdrag**

### CODE REVIEW

File: src/connection.cpp
Comment (mckennagalvin) [inline src/connection.cpp:142]: nice

File: src/connection.hpp
Comment (mckennagalvin) [inline src/connection.hpp:88]: does this necessarily have to be in an enum? I think we could just use a const int or something like that

**Review by mckennagalvin — APPROVED** [review]
👍

**Comment by mattdrag** [comment]
As of right now, the static handler needs some debugging, index.html doesn't seem to work and the paths are using full paths. Other than than, I left a few TODO's around, we need to implement james's registerer, and thats about all I can think of at the moment.

**Comment by mckennagalvin** [comment]
👍

**Comment by mckennagalvin** [comment]
Merging these three pull requests to add fixes/resolve conflicts got a little messy but I think everything's resolved now -- gonna go ahead and merge this one in!

**Comment by mckennagalvin** [comment]
Also for the future if we have a pull request that relies on changes from another pull request that hasn't been merged yet, we should maybe wait until that pull request is merged in or just add commits to the same branch as the other code we're relying on -- want to make pull requests as atomic as possible and avoid merge issues

**Pull request merged**
